src/dialects/abstract/query-generator.js

```diff
@@ -330,7 +330,21 @@ class QueryGenerator {
         // If no conflict target columns were specified, use the primary key names from options.upsertKeys
         const conflictKeys = options.upsertKeys.map(attr => this.quoteIdentifier(attr));
         const updateKeys = options.updateOnDuplicate.map(attr => `${this.quoteIdentifier(attr)}=EXCLUDED.${this.quoteIdentifier(attr)}`);
-        onDuplicateKeyUpdate = ` ON CONFLICT (${conflictKeys.join(',')}) DO UPDATE SET ${updateKeys.join(',')}`;
+
+        const hasConflictWhere
+          = this._dialect.supports.inserts.onConflictWhere
+          && options.conflictWhere;
```

Comment:
It's a change of convention but could we throw instead if `this._dialect.supports.inserts.onConflictWhere` is false but `options.conflictWhere` is set?

I'm slowly pushing the project in that direction to stop users from thinking the option is being used when it's ignored (and potentially enabled in a minor update which could then be a breaking change)

Reply:
done!